cartography/intel/gcp/__init__.py

```diff
@@ -1,12 +1,102 @@
+import googleapiclient.discovery
 import logging
+from collections import namedtuple
 
-from cartography.intel.gcp import crm
+from cartography.intel.gcp import crm, compute
 
 logger = logging.getLogger(__name__)
+Resources = namedtuple('Resources', 'crm_v1 crm_v2 compute')
+
+
+def _get_crm_resource_v1(credentials):
+    """
+    Instantiates a Google Compute Resource Manager v1 resource object to call the Resource Manager API.
+    See https://cloud.google.com/resource-manager/reference/rest/.
+    :param credentials: The GoogleCredentials object
+    :return: A CRM v1 resource object
+    """
+    # cache_discovery=False to suppress extra warnings.
+    # See https://github.com/googleapis/google-api-python-client/issues/299#issuecomment-268915510 and related issues
+    return googleapiclient.discovery.build('cloudresourcemanager', 'v1', credentials=credentials, cache_discovery=False)
+
+
+def _get_crm_resource_v2(credentials):
+    """
+    Instantiates a Google Compute Resource Manager v2 resource object to call the Resource Manager API.
+    We need a v2 resource object to query for GCP folders.
+    :param credentials: The GoogleCredentials object
+    :return: A CRM v2 resource object
+    """
+    return googleapiclient.discovery.build('cloudresourcemanager', 'v2', credentials=credentials, cache_discovery=False)
+
+
+def _get_compute_resource(credentials):
+    """
+    Instantiates a Google Compute resource object to call the Compute API. This is used to pull zone, instance, and
+    networking data. See https://cloud.google.com/compute/docs/reference/rest/v1/.
+    :param credentials: The GoogleCredentials object
+    :return: A Compute resource object
+    """
+    return googleapiclient.discovery.build('compute', 'v1', credentials=credentials, cache_discovery=False)
+
+
+def _initialize_resources(credentials):
+    """
+    Create namedtuple of all resource objects necessary for GCP data gathering.
+    :param credentials: The GoogleCredentials object
+    :return: namedtuple of all resource objects
+    """
+    return Resources(
+        crm_v1=_get_crm_resource_v1(credentials),
+        crm_v2=_get_crm_resource_v2(credentials),
+        compute=_get_compute_resource(credentials)
+    )
+
+
+def _sync_single_project(session, resources, project_id, gcp_update_tag, common_job_parameters):
+    """
+    Handles graph sync for a single GCP project.
+    :param session: The Neo4j session
+    :param resources: namedtuple of the GCP resource objects
+    :param project_id: The project ID number to sync.  See  the `projectId` field in
+    https://cloud.google.com/resource-manager/reference/rest/v1/projects
+    :param gcp_update_tag: The timestamp value to set our new Neo4j nodes with
+    :param common_job_parameters: Other parameters sent to Neo4j
+    :return: Nothing
+    """
+    compute.sync_gcp_instances(session, resources.compute, project_id, gcp_update_tag, common_job_parameters)
```

Comment:
See AWS module: log an INFO-level message that compute is being synced for this GCP project.

https://github.com/lyft/cartography/blob/master/cartography/intel/aws/__init__.py#L32